firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -62,18 +115,129 @@ public static FirebaseAppDistribution getInstance() {
    * starts an installation If the latest release is an AAB, directs the tester to the Play app to
    * complete the download and installation.
    *
-   * @throws an {@link Status.UPDATE_NOT_AVAILABLE_ERROR} exception if no new release is cached from
-   *     checkForUpdate
+   * @throws FirebaseAppDistributionException with UPDATE_NOT_AVAIALBLE exception if no new release
+   *     is cached from checkForUpdate
    */
   @NonNull
   public UpdateTask updateApp() {
     return (UpdateTask) Tasks.forResult(new UpdateState(0, 0, UpdateStatus.PENDING));
   }
 
+  private boolean supportsCustomTabs(Context context) {
+    Intent customTabIntent = new Intent("android.support.customtabs.action.CustomTabsService");
+    customTabIntent.setPackage("com.android.chrome");
+    List<ResolveInfo> resolveInfos =
+        context.getPackageManager().queryIntentServices(customTabIntent, 0);
+    return resolveInfos != null && !resolveInfos.isEmpty();
+  }
+
+  public static @NonNull String getApplicationName(@NonNull Context context) {
+    try {
+      return context.getApplicationInfo().loadLabel(context.getPackageManager()).toString();
+    } catch (Exception e) {
+      Log.e(TAG, "Unable to retrieve App name");
+      return "";
+    }
+  }
+
+  private void openSignIn(Uri uri) {
+    currentlySigningIn = true;
+    if (supportsCustomTabs(firebaseApp.getApplicationContext())) {
+      // If we can launch a chrome view, try that.
+      CustomTabsIntent.Builder builder = new CustomTabsIntent.Builder();
+
+      builder.setStartAnimations(
+          currentActivity.getApplicationContext(), R.anim.slide_in_right, R.anim.slide_out_left);
+
+      builder.setExitAnimations(
+          currentActivity.getApplicationContext(), R.anim.slide_in_left, R.anim.slide_out_right);
+
+      CustomTabsIntent customTabsIntent = builder.build();
+      Intent intent = customTabsIntent.intent;
+      intent.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY);
+      intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+
+      customTabsIntent.launchUrl(currentActivity, uri);
+
+    } else {
+      // If we can't launch a chrome view try to launch anything that can handle a URL.
+      Intent browserIntent = new Intent(Intent.ACTION_VIEW, uri);
+      ResolveInfo info = currentActivity.getPackageManager().resolveActivity(browserIntent, 0);
+      browserIntent.addFlags(Intent.FLAG_ACTIVITY_NO_HISTORY);
+      browserIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+      currentActivity.startActivity(browserIntent);
+    }
+  }
+
   /** Signs in the App Distribution tester. Presents the tester with a Google sign in UI */
   @NonNull
   public Task<Void> signInTester() {
-    return Tasks.forResult(null);
```

Comment:
Remove the comment here. 
Also should we consider adding some logic to handle the case where a previous signInTaskCompletionSource is not yet complete. Maybe if the previous`signInTaskCompleteionSource` is not complete, we just cancel it with an error and then proceed forward?

Reply:
Forgot to add cancellation logic, will do that in the next push.